[AMDGPU][Docs] DWARF aspace-aware base types

slinder1 opened the pull request:
Propose an extension to base type DIEs such that DW_ATE_address-encoded
base types can include an architecture specific address space. Use this
to implement DW_OP_convert conversions between AMDGPU address space
addresses where meaningful.

slinder1 committed:
[AMDGPU][Docs] DWARF aspace-aware base types

Propose an extension to base type DIEs such that DW_ATE_address-encoded
base types can include an architecture specific address space. Use this
to implement DW_OP_convert conversions between AMDGPU address space
addresses where meaningful.

t-tye commented on llvm/docs/AMDGPUUsage.rst:
```diff
@@ -2914,6 +2914,34 @@ controlled by the execution mask. An undefined location description together
+
+  Converting from ``DW_ASPACE_AMDGPU_generic`` to any of
+  ``DW_ASPACE_AMDGPU_local``, ``DW_ASPACE_AMDGPU_private_wave`` or
+  ``DW_ASPACE_AMDGPU_private_lane`` is defined when the relevant hardware
```
Doesn't it also require that the generic address is in the requested address space, otherwise it is an evaluation error?

slinder1 replied:
Yes, latest version should add these requirements

slinder1 committed:
Clean up requirements and fix typo

Describe the dynamic requirement that a generic address must be in the
destination address space for a conversion to be defined.
